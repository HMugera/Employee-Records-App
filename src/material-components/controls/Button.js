import React from "react";
import { Button as MuiButton, makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(0.5),
  },
  label: {
    textTransform: "none",
  },
}));

function Button({ text, size, color, variant, onClick, ...other }) {
  const classes = useStyles();
  return (
    <MuiButton
      classes={{ root: classes.root, label: classes.label }}
      variant={variant || "contained"}
      size={size || "large"}
      color={color || "primary"}
      onClick={onClick}
      {...other}
    >
      {text}
    </MuiButton>
  );
}

export default Button;
