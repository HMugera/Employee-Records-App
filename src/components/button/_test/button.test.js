import React from "react";
import Button from "./../Button";
import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button></Button>, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label='click me please'></Button>);
  expect(getByTestId("button")).toHaveTextContent("click me please");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button label='save'></Button>).toJSON();
  expect(tree).toMatchSnapshot(`
    <div
      className="button-style"
      data-testid="button"
    >
      save
    </div>
  `);
});
it("matches snapshot 2", () => {
  const tree = renderer
    .create(<Button label='click me please'></Button>)
    .toJSON();
  expect(tree).toMatchSnapshot(`
    <div
      className="button-style"
      data-testid="button"
    >
      save
    </div>
  `);
});
