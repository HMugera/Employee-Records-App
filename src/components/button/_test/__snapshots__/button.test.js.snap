// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 2: 
    <div
      className="button-style"
      data-testid="button"
    >
      save
    </div>
   1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  click me please
</div>
`;

exports[`matches snapshot: 
    <div
      className="button-style"
      data-testid="button"
    >
      save
    </div>
   1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  save
</div>
`;
